doc/bootloader/README.md

```diff
@@ -0,0 +1,113 @@
+
+You've probably guessed it, we chose the 2nd solution, and we decided to implement the OTA procedure in the application firmware. In this case, the bootloader is much simpler, and firmware developers have a lot more freedom in their design choices (like the BLE stack to integrate, the data transfer protocol to implement,...).
+
+Here is basically how it works:
+
+- the application firmware exposes a BLE "firmware update" service. In InfiniTime we decided to implement the **nRF DFU** protocol.
+- this update service stores the new firmware in a temporary location in the *external flash memory* and checks the integrity of the data (CRC).
+- when the new firmware is completely transferred and checked, the watch is rebooted and the bootloader is ran again. The bootloader detects that a new firmware is available in the *external* flash memory and safely swaps the current firmware in the *internal* memory with the new one from the *external* memory.
+- When the swapping is done and checked, the new firmware is executed from the *internal* memory.
+- If the new firmware does not properly run, the bootloader will automatically revert to the older firmware by swapping again firmwares from the external to the internal memory.
+
+This procedure involves multiple software modules and rely on some hardware features. Let's go into more details!
+
+### Memory map
+
+Multiple memories are available in the PineTime : the internal RAM (64KB) and flash (512KB) memories from the nRF52832 and an external flash memory (4MB) connected to the SPI bus.
+
+RAM memory is not relevant here. Here is how we organized both flash memories:
+
+![memory map](memoryMap.jpg)
+
+The internal flash memory contains the bootloader, a log segment (not used), the application firmware and a scratch area. This area is used by the bootloader when *swapping* firmware from the external memory.
+
+The SPI flash memory contains a recovery firmware and an area big enough to store a whole firmware (the new firmware will be first written to this area before being swapped into the internal memory). The remaining space is available for the application firmware needs.
+
+### The bootloader
+[The bootloader](https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader) is based on [MCUBoot, a secure bootloader for 32 bits microcontrollers](https://github.com/mcu-tools/mcuboot). It was originally written by [Lup Yuen](https://github.com/lupyuen).
+
+MCUBoot is run at boot time. In normal operation, it just jumps to the reset handler of the application firmware to run it. Once the application firmware is running, MCUBoot does not run at all.
+
+![MCUBoot boot sequence diagram](boot.png)
+
+But MCUBoot does much more than that : it can upgrade the firmware that is currently running by a new one, and it is also able to revert to the previous version of the firmware in case the new one does not run properly.
```

Comment:
replace `by a new one` with `with a new one`
```suggestion
But MCUBoot does much more than that:
it can upgrade the firmware that is currently running with a new one, and it is also able to revert to the previous version of the firmware in case the new one does not run properly.
```